### DISCUSSION TITLE
Using Hydra for integration/e2e tests for other Go projects

### DISCUSSION THREAD

**Discussion by simongottschlag** [ory/hydra#2628] [Q&A] [answered]
Hi!

I'm looking into building some tooling around OIDC and want to use a real provider for parts of my tests. It would be really cool if it was possible to start a provider directly in my Go tests.

Is there a documented way of how Ory could be used like this? I've been looking into how it's started with cobra but it's quite comprehensive and before I start doing anything I thought I should ask.

Thank you for the great work you are doing!

**Comment by aeneasr** [comment, marked as answer]
Unfortunately not but you could try out https://github.com/ory/dockertest - that’s how we do it in kratos

**Reply by simongottschlag** [reply]
@aeneasr thank you! If I was able to get something working with pure Go, would you be open to a PR? Maybe I could add something in the test-directory that exposed some easy to use API for unit/integration tests?

**Reply by simongottschlag** [reply]
To be honest, I spent a few hours trying to get it to work without success. Most likely due to me not having worked with the project enough - but since it's only for testing other projects I'll continue on the road of extending the go-oauth2-server with bare minimum to get by with OIDC.

I'll come back to Hydra in the future, but for now I've give up on trying to get it to work in-memory with Go.

Thank you for all the hard work you are doing!
